crates/env/src/api.rs

```diff
@@ -823,3 +823,42 @@ where
         TypedEnvBackend::call_runtime::<E, _>(instance, call)
     })
 }
+
+/// Adds a new delegate dependency to the contract.
```

Comment:
Need to add a notion of who can add/remove the lock on the code

Reply:
Anyone can (from the context of the calling contract).